### PR TITLE
[Spark] Prevent DeltaTableV2.toString from leaking catalog-injected storage credentials in plan trees and exceptions

### PR DESCRIPTION
**Pull request by seewishnew**
#### Which Delta project/connector is this regarding?

- [x] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description

`DeltaTableV2.toString` previously interpolated the full `CatalogTable`, `SparkSession`, and raw `options` map. When a catalog (e.g. Unity Catalog) injects Hadoop filesystem config into `CatalogTable.storage.properties` at table-load time, stringifying `CatalogTable` dumps `fs.*` keys — including auth tokens, access keys, and internal URIs — into exception messages, EXPLAIN output, and logs.

The primary vector: Spark's `ExtendedAnalysisException` appends the analyzed plan tree to `getMessage`. For unsupported operations like `SHOW CREATE TABLE` on Delta, `TreeNode.argString` calls `DeltaTableV2.toString` via `case other => other :: Nil` (since `DeltaTableV2` is a V2 `Table`, not a direct `CatalogTable` product field).

This PR keeps the original `toString` format with all six constructor parameters, but filters `fs.*` keys from both `CatalogTable.storage.properties` and the `options` map before stringifying. It reuses the existing `HIDDEN_STORAGE_PROPERTY_PREFIXES` constant from `DeltaTableV2.properties()` (introduced in PR #6300) as the single source of truth for which key prefixes to suppress.

Complements PR #6300 which filters `fs.*` from `DeltaTableV2.properties()` (the V2 DESCRIBE/SHOW TBLPROPERTIES path). That fix does not cover `toString`.

## How was this patch tested?

New `DeltaTableV2Suite` with 11 tests:

- **Negative (2):** Asserts `toString` omits all `fs.*` storage keys and values (6 key varieties); verifies `fs.*` keys are filtered while non-`fs` keys (e.g. `delta.*`, `io.unitycatalog.*`) are preserved.
- **Positive (4):** Verifies `CatalogTable` metadata (identifier, type, provider) is preserved; all six original parameters appear; path-only and tableIdentifier-only variants work correctly.
- **Options filtering (2):** Verifies `fs.*` keys in the `options` map are filtered; verifies simultaneous filtering of both `CatalogTable` storage and `options`.
- **Edge case (1):** Empty storage properties.
- **Integration (2):** Uses `DummySessionCatalog` (injects `fs.myKey` into storage properties). `EXPLAIN EXTENDED` test validates no plan node leaks credentials. `SHOW CREATE TABLE` test intercepts the `AnalysisException` and asserts both `getMessage` and `toString` omit storage properties — directly exercising the `ExtendedAnalysisException` → plan tree → `DeltaTableV2.toString` chain.

All 11 tests pass (`./build/sbt "spark/testOnly org.apache.spark.sql.delta.catalog.DeltaTableV2Suite"`).

## Does this PR introduce _any_ user-facing changes?

Yes. The string representation of `DeltaTableV2` in exception messages and plan output no longer includes `fs.*` storage properties or `fs.*` entries from the options map. The overall `toString` format is unchanged — all six constructor parameters are still present. This is a security improvement that prevents catalog-injected credentials from appearing in error messages, EXPLAIN output, and logs.

### CODE REVIEW

File: spark/src/test/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2Suite.scala [hunk 1]
```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
```
Comment (tdas) [inline spark/src/test/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2Suite.scala]: 2026

File: spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala [hunk 2]
```diff
@@ -431,8 +431,14 @@ class DeltaTableV2 private(
     deltaTableV2
   }
 
-  override def toString: String =
-    s"DeltaTableV2($spark,$path,$catalogTable,$tableIdentifier,$timeTravelOpt,$options)"
+  // Avoid stringifying CatalogTable or options (may contain fs.* credentials).
+  // Inlines the name() fallback using `path` to avoid triggering lazy deltaLog init.
```
Comment (tdas) [inline spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala]: i dont get .. what name are you talking about?

File: spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala [hunk 3]
```diff
@@ -431,8 +431,14 @@ class DeltaTableV2 private(
...
+  override def toString: String = {
+    val label = catalogTable.map(_.identifier.unquotedString)
+      .orElse(tableIdentifier)
+      .getOrElse(s"delta.`$path`")
+    s"DeltaTableV2($label, path=$path)"
```
Comment (tdas) [inline spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala]: Where happened to the other parameters that used to be printed before? why are you changing the format overall (you are introducing `path=xxx`)?

Comment (seewishnew) [reply spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala]: Addressed the extraneous `path=` bit here, but just to clarify the reasoning with removing the other constructor params that were logged out this is that:
1. `toString` is not a contract 
2. there's precedence with similar entities that compose catalogTable: [V1Table](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/V1Table.scala#L75), [DeltaSink](https://github.com/delta-io/delta/blob/37c540c78efbdda3283a8ac7203249deec0b026c/spark/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSink.scala#L262), [DeltaSource](https://github.com/delta-io/delta/blob/37c540c78efbdda3283a8ac7203249deec0b026c/spark/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSource.scala#L1227), [TahoeFileIndex](https://github.com/delta-io/delta/blob/37c540c78efbdda3283a8ac7203249deec0b026c/spark/src/main/scala/org/apache/spark/sql/delta/files/TahoeFileIndex.scala#L177-L181), all log out a qualified name and/or table path

File: spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala (hunk 2 above)
Comment (seewishnew) [reply spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala]: The existing [name()](https://github.com/delta-io/delta/blob/37c540c78efbdda3283a8ac7203249deec0b026c/spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala#L151-L153) method may end up making network calls in a [fallback path](https://github.com/delta-io/delta/blob/37c540c78efbdda3283a8ac7203249deec0b026c/spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala#L109-L131), which we'd typically want to avoid in a `toString` (which may be called in exception stack trace, etc.)

File: spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala [hunk 4]
```diff
@@ -431,8 +431,21 @@ class DeltaTableV2 private(
...
+  // to prevent catalog-injected or user-supplied credentials from leaking into exception messages,
+  // EXPLAIN output, and logs.
+  override def toString: String = {
+    val safeCatalogTable = catalogTable.map { ct =>
+      ct.copy(storage = ct.storage.copy(properties =
+        ct.storage.properties.filterNot { case (k, _) =>
+          DeltaTableV2.HIDDEN_STORAGE_PROPERTY_PREFIXES.exists(k.startsWith)
+        }))
+    }
+    val safeOptions = options.filterNot { case (k, _) =>
```
Comment (seewishnew) [inline spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala:444]: In the original format, this `$options`, on top of `$catalogTable.storage` properties is also a potential vector for leaking `fs.*` properties:

1. If `DeltaDataSource.createRelation` is called with `LOAD_FILE_SYSTEM_CONFIGS_FROM_DATAFRAME_OPTIONS` enabled
2. `DeltaTable.forPath(spark, path, hadoopConf)` is used

Comment (tdas) [reply spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala:444]: that makes sense.
